internal/venice-client-common/src/main/java/com/linkedin/venice/schema/merge/SortBasedCollectionFieldOpHandler.java

```diff
@@ -1070,4 +1067,29 @@ private boolean ignoreIncomingRequest(
     }
     return false;
   }
+
+  private boolean shouldUpdateMapFieldItemValueWithSameTs(Object currentValue, Object newValue, Schema fieldSchema) {
+    // This is a safeguard not to compare with null value.
+    if (currentValue == null) {
+      return true;
+    }
+    if (newValue == null) {
+      return false;
```

Comment:
Is it possible that these can be `null` since they are extracted from a map?
If yes, can you leave more javadoc to avoid future confusion?